### PR TITLE
Explore file-based metadata

### PR DESCRIPTION
**Pull request by mih**
At present just playing with toys to explore the problem space. 

Lives on top of #1074 for now.

### CODE REVIEW

File: datalad/metadata/parsers/fileinfo.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        ds_meta = _get_base_metadata_dict(dsid)
+        parts = []
+        if not self.has_metadata():
+            return meta
+        # shortcuts
+        cfg = self.ds.config
+        repo = self.ds.repo
+        cfg_section = self.cfg_section
+        files = repo.get_annexed_files()
+        # TODO RF to do this with one annex call
```
Comment (bpoldrack) [inline datalad/metadata/parsers/fileinfo.py]: Note: For this TODO, an issue would be nice, telling that we need AnnexRepo.get_file_key() to work on a list and in "batch mode".

**Comment by bpoldrack** [comment]
Do we consider files in annex only, because they have a key? Or is there another reason to exclude files in git?

File: datalad/metadata/parsers/fileinfo.py (hunk 1 above)
Comment (mih) [inline datalad/metadata/parsers/fileinfo.py]: There is #1076

**Comment by mih** [comment]
> Do we consider files in annex only, because they have a key? Or is there another reason to exclude files in git? 

This is just a simple thing to generate some file-based meta data. Many improvements are possible, but this PR is about figuring out the general issues and principles.

**Comment by mih** [comment]
Before someone is wondering. The "disable some parsers but guess the others" feature is already implemented. Will push when tests are done.

**Comment by mih** [comment]
Those are in already.

**Comment by mih** [comment]
Merging with #1074

**Pull request closed without merging**